python/guide/packages/optimizer/optimizer.md

```diff
@@ -0,0 +1,413 @@
+
+When training a deep learning model using the MXNet [gluon API](http://beta.mxnet.io/guide/packages/gluon/index.html), a gluon [Trainer](http://beta.mxnet.io/guide/packages/gluon/trainer.html) is initialized with the all the learnable parameters and the optimizer to be used to learn those parameters. A single step of iterative refinement of model parameters in MXNet is achieved by calling [`trainer.step`](http://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.Trainer.step.html) which in turn uses the gradient (and perhaps some state information) to update the parameters by calling `optimizer.update`. 
+
+Here is an example of how a trainer with an optimizer is created for, a simple Linear (Dense) Network.
+
+
+```python
+from mxnet import gluon, optimizer
+
+net = gluon.nn.Dense(1)
+net.initialize()
+optim = optimizer.SGD(learning_rate=0.1)
+trainer = gluon.Trainer(net.collect_params(), optimizer=optim)
+```
+
+In model training, the code snippet above would be followed by a training loop which, at every iteration performs a forward pass (to compute the loss), a backward pass (to compute the gradient of the loss with respect to the parameters) and a trainer step (which updates the parameters using the gradient). See the [gluon Trainer guide](http://beta.mxnet.io/guide/packages/gluon/trainer.html) for a complete example.
+
+We can also create the trainer by passing in the optimizer name and optimizer params into the trainer constructor directly, as shown below.
+
+
+```python
+trainer = gluon.Trainer(net.collect_params(), optimizer='adam', optimizer_params={'learning_rate':1})
+```
+
+### What should I use?
+For many deep learning model architectures, the `'sgd'` and `'adam'` optimizers are a really good place to start. If you are implementing a deep learning model and trying to pick an optimizer, start with [`'sgd'`](http://beta.mxnet.io/api/gluon-related/_autogen/mxnet.optimizer.SGD.html#mxnet.optimizer.SGD) as you will often get good enough results as long as your learning problem is tractable. If you already have a trainable model and you want to improve the convergence then you can try [`'adam'`](http://beta.mxnet.io/api/gluon-related/_autogen/mxnet.optimizer.SGD.html#mxnet.optimizer.Adam). If you would like to improve your model training process further, there are a number of specialized optimizers out there with many of them already implemented in MXNet. This guide walks through these optimizers in some detail.
```

Comment:
Would remove pips in this sections e.g. `'sgd'` -> `sgd`

Reply:
addressed.